src/pybind/mgr/cephadm/services/container.py

```diff
@@ -14,6 +16,14 @@ class CustomContainerService(CephadmService):
     def prepare_create(self, daemon_spec: CephadmDaemonSpec[CustomContainerSpec]) \
             -> CephadmDaemonSpec:
         assert self.TYPE == daemon_spec.daemon_type
+        if daemon_spec.spec is None:
+            # Exit here immediately because the required service
+            # spec to create a daemon is not provided. This is only
+            # provided when a service is applied via 'orch apply'
+            # command.
+            msg = "Required service specification not provided"
+            raise OrchestratorError(msg)
```

Comment:
I'm disagreeing with @liewegas here, but IMO we should simplify things and *only* allow daemons with specs.